addons/mail/static/src/core/common/suggestion_hook.js

```diff
@@ -171,6 +183,22 @@ export class UseSuggestion {
         this.clearSearch();
         this.composer.insertText(`${option.label} `, position);
     }
+    insertHtml(option) {
+        const { startContainer, endContainer, endOffset } =
+            this.comp.editor.shared.selection.getEditableSelection();
+        const range = {
+            startContainer,
+            startOffset: this.search.position,
+            endContainer,
+            endOffset,
+        };
+        this.comp.editor.shared.delete.deleteRange(range);
+        if (option.cannedResponse) {
+            this.composer.cannedResponses.push(option.cannedResponse);
+        }
```

Comment:
I don't like that we duplicate this part.

I'd rather we keep a singe `insert` method, and add conditions for `composerService.htmlEnabled` inside it.